checkout/views.py

```diff
@@ -0,0 +1,190 @@
+			})
+		except UserProfile.DoesNotExist:
+			order_form = OrderForm()
+
+	stripe_public_key = settings.STRIPE_PUBLIC_KEY
+
+	if not stripe_public_key:
+		messages.warning(request, 'Stripe public key is missing. Did you forget to set it in your environment?')
+		
+	template = 'checkout/checkout.html'
+	context = {
+		'stripe_public_key': stripe_public_key,
+		'order_form': order_form,
+	}
+
+	return render(request, template, context)
+
+
+def checkout_success(request, order_number):
+	# Check the session to see if the user wanted to save their info
+	save_info = request.session.get('save_info')
+
+	order = Order.objects.get(order_number=order_number)
+	no_errors = True
+
+	# Attach the order to the user's profile
+	# Handle issues w/ appropriate warnings
+	try:
+		user_profile = UserProfile.objects.get(user=request.user)
+
+		# User the standard UserProfileForm to save to profile
+		if save_info:
+			form_data = {
+				'default_phone_number': order.phone_number,
+				'default_country': order.country,
+				'default_postcode': order.postcode,
+				'default_town_or_city': order.town_or_city,
+				'default_street_address1': order.street_address1,
+				'default_street_address2': order.street_address2,
+				'default_county': order.county,
+			}
+			user_profile_form = UserProfileForm(form_data, instance=user_profile)
+			if user_profile_form.is_valid():
+				user_profile_form.save()
+			else:
+				no_errors = False
+				messages.warning(request, (
+					f'Your order was successfully processed, but your information '
+					f'could not be saved to your profile. Please try adding it '
+					f'manually on your profile page. Your order number is {order_number}. '
+					f'A confirmation email has been sent to {order.email}.')
+				)
+		# Now attempt to attach the order to the user's profile
+		try:
+			order.user_profile = user_profile
+			order.save()
+		except:
+			no_errors = False
+			messages.warning(request, (
+				f'Your order was successfully processed, but you may not see this '
+				f'order in your order history in your profile due to an error attaching '
+				f'the order to your profile. Please contact us for any assistance '
+				f'needed with this order. Your order number is {order_number}. '
+				f'A confirmation email has been sent to {order.email}.')
+			)
+	except UserProfile.DoesNotExist:
+		no_errors = False
+		messages.warning(request, (
+			f'Your order was successfully processed, but your information '
+			f'could not be saved to your profile because your profile was '
+			f'not found. Please contact us if you would like to save your '
+			f'information for future orders. Your order number is {order_number}. '
+			f'A confirmation email has been sent to {order.email}.')
+		)
+
+	# If there were no issues, provide a standard success message
+	if no_errors:
+		messages.success(request, f'Order successfully processed! Your order number is {order_number}. A confirmation email has been sent to {order.email}.')
```

Comment:
I feel uneasy saying to the user that it's been sent before it has - there could be other errors in between

Reply:
Fair point! I will extract the email bit out into a separate function, then conditionally send this message based on the result, or do it in the function itself.